test/cookbooks/keepalived_test/recipes/fake_interface.rb

```diff
@@ -0,0 +1,53 @@
+#
+# Cookbook:: keepalived_test
+# Recipe:: eth1
+#
+# Copyright:: 2018, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def make_dummy(interface, ipv4, ipv6 = nil)
```

Comment:
I'm confused why you have a ruby function defined in a recipe even if this is a test cookbook. Why are you doing this? Seems like there would be a better way.

Reply:
because this code is duplicated with slight variations by several of the suites so I offloaded it to a helper function then call that function in the test recipes

Reply:
This shouldn't be in a recipe however. Would it make more sense to just do this with attributes and set them as you need?

Reply:
alright I'm not sure if it's exactly how you wanted it but representing all the attributes in yaml is a bit beyond my comprehension so I set the attributes in the recipe and include the fake interface.